Update Forcing-data-models.md (#1391)
COSIMA/MOM6-CICE6: documentation/docs/pages/inputs/Forcing-data-models.md

1. Shortwave radiation, captured by `SW`
2. Longwave radiation, captured by `LW`
3. Latent heat, captured by `latent`. This is the sum of latent heat from evaporation `latent_evap`, latent heat from frozen precipitation `latent_fprec_diag`, latent heat from frozen runoff `latent_frunoff` and latent heat from frozen runoff from glaciers `latent_frunoff_glc` (which is zero in ACCESS-OM3, since we don't couple to an active ice sheet model).    
4. Sensible heat, captured by `sensible`
5. Heat from melt/freezing of sea ice, captured by `seaice_melt_heat`
6. Heat from frazil formation, captured by `frazil`
7. Heat from mass transfer, frozen or liquid, given by `heat_content_surfwater`, which can be further partitioned into:  
   - `heat_content_evap`, from evaporation
   - `heat_content_cond`, from condensation
   - `hfrainds`, from liquid and frozen precipitation, further split into (i) `heat_content_lprec` and (ii) `heat_content_fprec`
   - `hfrunuoffds` from liquid and frozen runoff, further split into (i) `heat_content_lrunoff` and (ii) `heat_content_frunoff`   

> 8. Heat from flux adjustments, captured by `heat_added`.

Reviewer: ok - to avoid confusion I suggest
```suggestion
8. Heat from flux adjustments, captured by `heat_added` (zero in ACCESS-OM3).
```

This [iterates towards convergence of `ustar`](https://github.com/ESCOMP/CMEPS/blob/bc29792d76c16911046dbbfcfc7f4c3ae89a6f00/cesm/flux_atmocn/shr_flux_mod.F90#L393) to a relative error of less than [`flux_convergence = 0.01`](https://github.com/search?q=repo%3AACCESS-NRI%2Faccess-om3-configs+path%3Adoc%2Fnuopc.runconfig+flux_convergence&type=code), if this can be achieved in [`flux_max_iteration = 5`](https://github.com/search?q=repo%3AACCESS-NRI%2Faccess-om3-configs+path%3Adoc%2Fnuopc.runconfig+flux_max_iteration&type=code) iterations or fewer. The atmosphere-ocean stress is [calculated using the relative wind](https://github.com/ESCOMP/CMEPS/blob/bc29792d76c16911046dbbfcfc7f4c3ae89a6f00/cesm/flux_atmocn/shr_flux_mod.F90#L434-L438), i.e. the difference between the surface wind and surface current velocity.

## Ocean surface fluxes

### Salt fluxes

> The net salt flux across the ocean's surface is given by the `salt_flux` diagnostic. Salt fluxes are associtated to two different processes:

Reviewer: ```suggestion
The net salt flux across the ocean's surface is given by the `salt_flux` diagnostic. Salt fluxes are associated with two different processes:
```

1. Salt fluxes from sea ice, captured by the `salt_flux_in` diagnostic.

> 2. Salt fluxes from restoring, captured by `salt_flux_added`. Note that restoring can be applied as either salt or freshwater fluxes, which is determined in the `MOM_input` file.

Reviewer 2: Should we say something like, in these configs, we apply restoring as a salt flux, but it could be configured as a freshwater flux in `MOM_input` ?

Author: Should I rephrase this: "Note that restoring can be applied as either salt or freshwater fluxes, which is determined in the `MOM_input` file."?

Reviewer: ```suggestion
2. Salt fluxes from sea surface salinity restoring, captured by `salt_flux_added` (note that SSS restoring is applied as a salt (rather than freshwater) flux in ACCESS-OM3, because we set `SRESTORE_AS_SFLUX = True` in the `MOM_input` file).
```

### Freshwater fluxes

> The net freshwater flux across the ocean's surface is given by either the `wfo` or `PRCmE` diagnostics. Freshwater fluxes are associated to six different processes:

Reviewer 2: ```suggestion
The net freshwater flux across the ocean's surface is given by the `wfo` diagnostics (sometimes the native MOM name `PRCmE` is used). Freshwater fluxes are associated to six different processes:
```

> The net freshwater flux across the ocean's surface is given by the `wfo` diagnostics (sometimes the native MOM name `PRCmE` is used). Freshwater fluxes are associated to six different processes:

Reviewer: ```suggestion
The net freshwater flux across the ocean's surface is given by the `wfo` diagnostic (sometimes the native MOM name `PRCmE` is used). Freshwater fluxes are associated with six different processes:
```

1. Precipitation, captured by the `precip` diagnostic. Total precipitation is the sum of: 
   - Liquid precipitation, captured by `lprec`

>    - Frozen precipitation, captured by `fprec`

Reviewer 2: This didn't format right

<img width="633" height="159" alt="Image" src="https://github.com/user-attachments/assets/7b4e6cbb-99e9-43be-82d8-a2b9bc6e280b" />

https://access-om3-configs--1391.org.readthedocs.build/1391/inputs/Forcing-data-models/#ocean-surface-fluxes

Reviewer: ```suggestion
1. Total precipitation, captured by the `precip` diagnostic. This is the sum of liquid precipitation `lprec` and frozen precipitation `fprec`.
```

Reviewer 3: > This didn't format right
> 
> <img alt="Image" width="633" height="159" src="https://private-user-images.githubusercontent.com/79179784/601247915-7b4e6cbb-99e9-43be-82d8-a2b9bc6e280b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODA5NzMyOTgsIm5iZiI6MTc4MDk3Mjk5OCwicGF0aCI6Ii83OTE3OTc4NC82MDEyNDc5MTUtN2I0ZTZjYmItOTllOS00M2JlLTgyZDgtYTJiOWJjNmUyODBiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA2MDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNjA5VDAyNDMxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ0NzYzZTA1YmQ5MTJkN2ZmZTcyMTJlMWQwMWE3MDcyNTkzNzA2MDUwMjAzMDgwYThiMDJkMWFhNzExNmYyMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.YakGkv88kSsTwA-82yUzHdXViwtCG6iEnd_EBACqGRA">
> https://access-om3-configs--1391.org.readthedocs.build/1391/inputs/Forcing-data-models/#ocean-surface-fluxes

At a guess you need a space inbetween 
>  Total precipitation is the sum of: 

insert line break

>  - Liquid precipitation, captured by `lprec`

Might be [relevant here too](https://github.com/ACCESS-NRI/access-om3-configs/pull/1391#discussion_r3338035278). (@julia-neme mkdoc markdown is slightly different to Github's)

> 2. Evaporation, captured by `evap`

Reviewer 2: ```suggestion
2. Evaporation, captured by `evap`. (This also includes sublimation, condensation and deposition).
```

3. Liquid runoff, captured by `lrunoff` or `friver`
4. Frozen runoff, captured by `frunoff` or `ficeberg`
5. Sea ice melt/formation, captured by `seaice_melt`

> 6. Virtual precipitation from salinity restoring, captured by `vprec`

Reviewer 2: `lprec`, `fprec` and `evap` only capture the ocean fraction. There are similar sea-ice diagnostics, (`frain, `fsnow`, `evap`) 

`seaice_melt` matches the CICE `fresh` diagnostic

Author: Yeah, I focused on "Ocean surface fluxes", but there could be another section on "Sea ice fluxes"? Although I haven't explored these at all.

Reviewer: Yes at some point we should also do an equivalent effort at documenting CICE water, salt and heat flux diagnostics.

> 5. Sea ice melt/formation, captured by `seaice_melt`

Reviewer: ```suggestion
5. Sea ice melt/formation, captured by `seaice_melt` (same as the CICE diagnostic `fresh`)
```

> 6. Virtual precipitation from salinity restoring, captured by `vprec`

Reviewer: ```suggestion
6. Virtual precipitation from salinity restoring, captured by `vprec` (zero in ACCESS-OM3, because we use `SRESTORE_AS_SFLUX = True`)
```
(have I got this correct?)

Author: No, it is non-zero. Varies with time but is spatially a constant

Reviewer: ok, so I guess it's calculated from the salt restoring flux then? So how about this? Hopefully correct, but I haven't checked the code
```suggestion
6. Virtual precipitation from salinity restoring, captured by `vprec` (this is not an actual freshwater input in ACCESS-OM3 because we use `SRESTORE_AS_SFLUX = True`; the values are calculated from the salt restoring)
```

Reviewer: Also, I don't think it will be spatially uniform in runs that use ADJUST_NET_SRESTORE_BY_SCALING = True

Reviewer 4: In the configuration @julia-neme was running, `vprec` gives the adjustment applied by `ADJUST_NET_FRESH_WATER_TO_ZERO=True`. It's calculated from the net fresh water, not the salt restoring.

We [no longer](https://github.com/ACCESS-NRI/access-om3-configs/pull/1399) set `ADJUST_NET_FRESH_WATER_TO_ZERO=True`, so `vprec` will be zero in future runs. I think we can just remove this bullet point.

Reviewer 2: ok removed vprec

There are also two other summarising diagnostics, `net_massin` and `net_massout` which contain the net mass of freshwater into and out of the ocean respectively. Note that these are made up of a combination of the individual fluxes described above, but that `seaice_melt` contributes to both depending on whether ice is formed or melted.
   
### Heat fluxes

The net heat flux across the ocean's surface is given by either `hfds` or `net_heat_surface`. Heat fluxes are associated to eight different processes:

1. Shortwave radiation, captured by `SW`
2. Longwave radiation, captured by `LW`
3. Latent heat, captured by `latent`. Latent heat can be further partitioned into:
   - `latent_evap`, latent heat from evaporation
   - `latent_fprec_diag`, latent heat from frozen precipitation
   - `latent_frunoff`, latent heat from frozen runoff

>    - `latent_frunoff_glc`, latent heat from frozen runoff from glaciers   

Reviewer 2: Do we ever use `frunoff_glc` - how is it different to `frunoff` ?

Author: Yeah, I don't know if we use `frunoff_glc` because it is zero everywhere, but thus it is not the same as `frunoff`:

<img width="1177" height="289" alt="image" src="https://github.com/user-attachments/assets/01c102fa-fb0c-41bf-bf82-6306c987c57c" />

So I thought to include it in the list because maybe at some point it does get used? Happy to delete.

Reviewer: My understanding is this is zero unless we couple to an active ice sheet model

>    - `latent_frunoff_glc`, latent heat from frozen runoff from glaciers   

Reviewer 2: This formatting need fixing as well - like the above one

Author: Mmmm I don't know how to fix these because they format fine in github.

>    - `latent_frunoff_glc`, latent heat from frozen runoff from glaciers   

Reviewer: ```suggestion
3. Latent heat, captured by `latent`. This is the sum of latent heat from evaporation `latent_evap`, latent heat from frozen precipitation `latent_fprec_diag`, latent heat from frozen runoff `latent_frunoff` and latent heat from frozen runoff from glaciers `latent_frunoff_glc` (which is zero in ACCESS-OM3, since we don't couple to an active ice sheet model).    
```

4. Sensible heat, captured by `sensible`
5. Heat from melt/freezing of sea ice, captured by `seaice_melt_heat`
6. Heat from frazil formation, captured by `frazil`

> 7. Heat from mass transfer, frozen or liquid, given by `heat_content_surfwater`, which can be further partitioned into:  

Reviewer: again formatting isn't working right here

   - `heat_content_evap`, from evaporation
   - `heat_content_cond`, from condensation
   - `hfrainds`, from liquid and frozen precipitation, further split into (i) `heat_content_lprec` and (ii) `heat_content_fprec`

>    - `hfrunuoffds` from liquid and frozen runoff, further split into (i) `heat_content_lrunoff` and (ii) `heat_content_frunoff`   

Reviewer 2: Do we use `hfrunuoffds` ? Is this just heat associated with the mass flux (assuming the freshwater comes in at the ocean temperature?)

Author: Not sure what it means if we use it or not sorry, but it is a diagnostic you can output that represents "Heat content (relative to 0C) of liquid+solid runoff into ocean" and is equal to the sum of `heat_content_lrunoff` and `heat_content_frunoff`. 

Note that it is only the mass flux from runoff, the mass flux from precipitation (liquid and frozen) comes in with `hfrainds`. On this note though, even though `hfrainds` says it is precipitation (liquid and frozen), it also includes heat from condensation:

<img width="1177" height="620" alt="image" src="https://github.com/user-attachments/assets/3100b099-00dd-464b-bd89-afa3ff28969d" />

> 8. Heat from flux adjustments, captured by `heat_added`.

Reviewer: zero in ACCESS-OM3, right?

Author: Yes - I included it because I assumed its something a user might decide to turn on?

1. Salt fluxes from sea ice, captured by the `salt_flux_in` diagnostic in MOM6 and the `fsalt` diagnostic in CICE.
2. Salt fluxes from sea surface salinity restoring, captured by `salt_flux_added` (note that SSS restoring is applied as a salt (rather than freshwater) flux in ACCESS-OM3, because we set `SRESTORE_AS_SFLUX = True` in the `MOM_input` file).

### Freshwater fluxes

Freshwater fluxes with the data models (DATM and DROF) are separated in ocean, sea ice and land components as follows. 

- **Precipitation:** precipitation over ocean and sea ice cells are scaled by the sea ice concentration in that cell. Therefore where sea ice covers part of a grid cell, a fraction of precipitation is received by each of the ocean and sea ice components, according to the sea ice concentration in that cell. Precipitation over land is discarded.
- **Evaporation:** evaporation (including sublimation, condensation and deposition) is calculated internally by CICE for sea ice. The mediator (CMEPS) calculates evaporation for the ocean and provides this as a surface forcing to MOM6.

> - **Runoff:** runoff only enters the Ocean. Therefore 100% of the runoff from DROF goes into MOM6.

Reviewer 4: ```suggestion
- **Runoff:** runoff only enters the ocean. Therefore 100% of the runoff from DROF goes into MOM6.
```

> The net freshwater flux across the ocean (MOM6) surface is given by the `wfo` diagnostic (sometimes the native MOM name `PRCmE` is used). Freshwater fluxes are associated with six different processes:

Reviewer 4: ```suggestion
The net freshwater flux across the ocean (MOM6) surface is given by the `wfo` diagnostic (sometimes the native MOM name `PRCmE` is used). Freshwater fluxes are associated with five different processes:
```

1. Total precipitation, captured by the `precip` diagnostic. This is the sum of liquid precipitation `lprec` and frozen precipitation `fprec`.
2. Evaporation, captured by `evap`.
3. Liquid runoff, captured by `lrunoff` or `friver`
4. Frozen runoff, captured by `frunoff` or `ficeberg`
5. Sea ice melt/formation, captured by `seaice_melt` (same as the CICE diagnostic `fresh`)

There are also two other summarising diagnostics, `net_massin` and `net_massout` which contain the net mass of freshwater into and out of the ocean respectively. Note that these are made up of a combination of the individual fluxes described above, but that `seaice_melt` contributes to both depending on whether ice is formed or melted.

The net freshwater flux across the sea ice (CICE) top surface is the sum of:

1. Precipitation: captured by `rain_ai` and `snow_ai`
2. Evaporation: captured by `evap_ai`.

> and the net freswater flux across the sea ice - ocean interface is `seaice_melt` / `fresh`.

Reviewer 4: ```suggestion
and the net freshwater flux across the sea ice - ocean interface is `fresh` (MOM6 diagnostic `seaice_melt`).
```

- **Latent Heat (Evaporation):** Latent heat due to evaporation are calculated internally by CICE for sea ice. The mediator (CMEPS) calculates these for the ocean and provides this as a surface forcing to MOM6.
- **Latent Heat (Melt):** The latent heat to melt frozen runoff (i.e. icebergs) is calculated internally in MOM6.
- **Heat from mass transfer:** The enthalpy transferred in/out of the ocean due to freshwater transfer is calculated in the mediator (CMEPS) and provided as a surface forcing to MOM6. For CICE6, this enthalpy flux is calculated internally.
- **Precipitation and Runoff:** Heat fluxes associated with transfer of freshwater are calculated in the relevant model components, for both MOM6 and CICE, based on the freshwater received by that component.

There is no prognostic process to raise/lower the temperature of freshwater fluxes to the ocean / sea ice temperature. It is assumed to enter the model at the temperature of the surrounding ocean / sea ice. There is also no process (currently) to account for latent heat to melt liquid runoff from icesheets (i.e. basal melt) - see [1197](https://github.com/ACCESS-NRI/access-om3-configs/issues/1197) for details.

> The net heat flux across the ocean's surface is given by either `hfds` or `net_heat_surface`. Heat fluxes are associated to eight different processes:

Reviewer 4: ```suggestion
The net heat flux across the ocean's surface is given by the `hfds` diagnostic (or native MOM name `net_heat_surface`). Heat fluxes are associated to eight different processes:
```

> 1. Shortwave radiation, captured by `SW`

Reviewer 4: ```suggestion
1. Shortwave radiation, captured by `rsntds` (or native MOM name `SW`)
```

> 2. Longwave radiation, captured by `LW`

Reviewer 4: ```suggestion
2. Longwave radiation, captured by `rlntds` (or native MOM name `LW`)
```

> 3. Latent heat, captured by `latent`. This is the sum of latent heat from evaporation `latent_evap`, latent heat from frozen precipitation `latent_fprec_diag`, latent heat from frozen runoff `latent_frunoff` and latent heat from frozen runoff from glaciers `latent_frunoff_glc` (which is zero in ACCESS-OM3, since we don't couple to an active ice sheet model).

Reviewer 4: ```suggestion
3. Latent heat, captured by `hflso` (or native MOM name `latent`). This is the sum of latent heat from evaporation `latent_evap`, latent heat from frozen precipitation `hfsnthermds` (or native MOM name `latent_fprec_diag`), latent heat from frozen runoff `hfibthermds` (or native MOM name `latent_frunoff`) and latent heat from frozen runoff from glaciers `latent_frunoff_glc` (which is zero in ACCESS-OM3, since we don't couple to an active ice sheet model).
```

> 4. Sensible heat, captured by `sensible`

Reviewer 4: ```suggestion
4. Sensible heat, captured by `hfsso` (or native MOM name `sensible`)
```

5. Heat from melt/freezing of sea ice, captured by `seaice_melt_heat` (same as CICE diagnostic `fhocn_ai`)

> 6. Heat from frazil formation, captured by `frazil`

Reviewer 4: ```suggestion
6. Heat from frazil formation, captured by `hfsifrazil` (or native MOM name `frazil`)
```

7. Heat from mass transfer, frozen or liquid, given by `heat_content_surfwater`, which can be further partitioned into:
      - `heat_content_evap`, from evaporation
      - `heat_content_cond`, from condensation

>       - `hfrainds`, from liquid and frozen precipitation, further split into (i) `heat_content_lprec` and (ii) `heat_content_fprec`

Reviewer 4: `hfrainds` [includes condensation](https://github.com/ACCESS-NRI/MOM6/blob/cf8bb6375aa2fc3ad4d5468df3aa9b1b9fd33d34/src/core/MOM_forcing_type.F90#L3199-L3207)

```suggestion
      - `hfrainds`, from liquid and frozen precipitation and condensation, further split into `heat_content_lprec`, `heat_content_fprec` and `heat_content_cond`
```

>       - `hfrunoffds` from liquid and frozen runoff, further split into (i) `heat_content_lrunoff` and (ii) `heat_content_frunoff`

Reviewer 4: ```suggestion
      - `hfrunoffds` from liquid and frozen runoff, further split into `heat_content_lrunoff`, `heat_content_frunoff`, `heat_content_lrunoff_glc` and `heat_content_frunoff_glc` (again, the `_glc` components are zero in ACCESS-OM3)
```
